src/main/kotlin/com/trackforever/controllers/ProjectController.kt

```diff
@@ -215,4 +296,34 @@ class ProjectController (@Autowired private var projectRepository: ProjectReposi
         }
     }
 
+    /**
+     * Generates the SHA3 hash from the given TrackForeverProject or TrackForeverIssue.
+     * Ignores the current hash field when performing calculations and issues (if project).
+     */
+    fun generateHash(jsonData: Any): String {
+        val mapper = jacksonObjectMapper()
+        when (jsonData) {
+            is TrackForeverProject -> {
+                val module = SimpleModule("TrackForeverProjectSerializer", Version(1, 0, 0, null, null, null))
+                module.addSerializer(TrackForeverProject::class.java, TrackForeverProjectSerializer())
+                mapper.registerModule(module)
+                val projectJson = mapper.writeValueAsString(jsonData)
+                logger.debug("Project JSON: $projectJson")
+                val crypto = SHA3.Digest512()
+                val projectByteArray = crypto.digest(projectJson.toByteArray())
+                return Hex.toHexString(projectByteArray)
+            }
+            is TrackForeverIssue -> {
+                val module = SimpleModule("TrackForeverIssueSerializer", Version(1, 0, 0, null, null, null))
+                module.addSerializer(TrackForeverIssue::class.java, TrackForeverIssueSerializer())
+                mapper.registerModule(module)
+                val issueJson = mapper.writeValueAsString(jsonData)
+                logger.debug("Issue JSON: $issueJson")
+                val crypto = SHA3.Digest512()
+                val issueByteArray = crypto.digest(issueJson.toByteArray())
+                return Hex.toHexString(issueByteArray)
+            }
+            else -> return ""
```

Comment:
You can just throw an error here. `throw Error("Should be given a TrackForever Issue or Project")` or something like that